vm/src/function/builtin.rs

```diff
@@ -3,9 +3,10 @@ use crate::{
     convert::ToPyResult, object::PyThreadingConstraint, PyPayload, PyRef, PyResult, VirtualMachine,
 };
 use std::marker::PhantomData;
+use std::sync::Arc;
 
 /// A built-in Python function.
-pub type PyNativeFunc = Box<py_dyn_fn!(dyn Fn(&VirtualMachine, FuncArgs) -> PyResult)>;
+pub type PyNativeFunc = Arc<py_dyn_fn!(dyn Fn(&VirtualMachine, FuncArgs) -> PyResult)>;
```

Comment:
I had to do this to make PyNativeFuncDef clonable, but actually it is a very wide change and can cause performance issues. (Arc::clone is called whenever calling native methods.)

Reply:
what if we keep PyFunction owns `Box<>` but PyBoundMethod holds `&'static`?

Reply:
Oh, I got what you meant. Maybe I can use Box::leak?

Reply:
not 100% sure, but it can be one of the way. do we always use static functions, never clousure? In that case, `leak` is one of the way.
I was thinking giving unbound method (e.g. `object.__repr__`) Box but bound method (`object().__repr__` reference)

Reply:
ah, i was totally not getting how this is going.

Reply:
@qingshi163 @DimitrisJim @coolreader18 any idea about this?